torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1237,23 +1237,16 @@ void ProcessGroupNCCL::heartbeatMonitor() {
   // local disk)
   std::future<bool> asyncDebugDump = launchAsyncDebugDump();
 
-  // Create a error message reported from MonitorThread, so
-  // we throw exception and make the whole process to be killed.
-  // TODO(fduwjj): After having a hang debug wiki, we need to update the wiki
-  // url here.
-  const auto exitMsg = c10::str(
+  const auto logMsg = c10::str(
       logPrefix(),
-      "ProcessGroupNCCL's watchdog got stuck for ",
-      heartbeatTimeoutInSec_,
-      "seconds without making progress in monitoring enqueued collectives. ",
-      "This typically indicates a NCCL/CUDA API hang blocking the watchdog, ",
-      "and could be triggered by another thread holding the GIL inside a ",
-      "CUDA api, or other deadlock-prone behaviors.",
-      "If you suspect the watchdog is not actually stuck and a longer timeout would help, ",
-      "you can either increase the timeout (TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC) to a larger value "
-      "or disable the heartbeat monitor (TORCH_NCCL_ENABLE_MONITORING=0)."
-      "If either of aforementioned helps, feel free to file an issue to PyTorch about the short timeout "
-      "or false positive abort; otherwise, please attempt to debug the hang.");
+      "Heartbeat monitor timed out! Process will be terminated after dumping debug info.",
+      " terminateProcessGroup_=",
+      terminateProcessGroup_,
+      ", collectiveDebugInfoMode_=",
+      collectiveDebugInfoMode_,
```

Comment:
Hmmm I am not sure if we want to log these two flags here. Maybe only workMetaList size?

Reply:
agree, i will remove them then. i did not add them (i was wondering myself whether they are useful) but decided not to make a change.